src/plugins/data/common/autocomplete_provider/index.ts

```diff
@@ -16,13 +16,22 @@
  * specific language governing permissions and limitations
  * under the License.
  */
+import { AutocompleteProvider } from './types';
 
-const autocompleteProviders = new Map();
+export class AutocompleteProviderRegister {
+  private readonly registeredProviders: Map<string, AutocompleteProvider> = new Map();
 
-export function addAutocompleteProvider(language, provider) {
-  autocompleteProviders.set(language, provider);
-}
+  public addProvider(language: string, provider: AutocompleteProvider): void {
+    if (language && provider) {
+      this.registeredProviders.set(language, provider);
+    }
+  }
+
+  public getProvider(language: string): AutocompleteProvider | undefined {
+    return this.registeredProviders.get(language);
+  }
 
-export function getAutocompleteProvider(language) {
-  return autocompleteProviders.get(language);
+  public clearProviders(): void {
```

Comment:
Is this ever used?

Reply:
@lizozom  I've execute this method from plugin.stop hook.